pkg/plugins/manager.go

```diff
@@ -11,160 +12,104 @@ import (
+	ErrUninitializedManager  = errors.New("uninitialized plugins manager")
 )
 
-// PluginType specifies the type of related plugins
-// For example, "diff" is a type of plugins that know how to perform a diff.
-type PluginType string
-
-// PluginName specifies the name of a plugin
-// For example, "delta" is a plugin.
-type PluginName string
+var allowedProtocols = []plugin.Protocol{
+	plugin.ProtocolGRPC,
+}
 
-// PluginIdentity identifies the plugin implementation.
+// PluginIdentity identifies the plugin's version and executable location.
 type PluginIdentity struct {
-	Impl               plugin.Plugin
-	ImplName           string
-	HandshakeConfig    plugin.HandshakeConfig
+	Version            int
 	ExecutableLocation string
 }
-type pluginTypeConfigMap map[PluginName]plugin.ClientConfig
 
-// PluginTypeNameMap example:
-/*
-{
-	"diff": {
-		"delta": plugin.clientConfig{
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/diff/delta"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"deltaDiff": DeltaGRPCPlugin,
-			},
-		},
-		"iceberg: {
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/diff/iceberg"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"icebergDiff": IcebergGRPCPlugin
-			},
-		}
-	},
-	"merge": {
-		"delta": {
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/merge/delta"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"deltaMerge": DeltaGRPCPlugin,
-			},
-		},
-		"iceberg: {
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/merge/iceberg"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"icebergMerge": IcebergGRPCPlugin
-			},
-		}
-	},
+// PluginAuth includes authentication properties for the plugin.
+type PluginAuth struct {
+	Key   string
+	Value string
 }
-*/
-type PluginTypeNameMap map[PluginType]pluginTypeConfigMap
 
-// The Manager holds the different types of plugins that can be used in the plugin system
-type Manager struct {
-	pluginTypes PluginTypeNameMap
+// The Manager holds a map for different kinds of possible plugin clients.
+// For example, the clients map might contain a mapping of "delta" -> plugin.Client to communicate with the Delta
+// plugin.
+type Manager[T any] struct {
```

Comment:
What is T?  `plugin.Client` is not parametrized on it, which makes it less clear.